keps/6757-failure-recovery/README.md

```diff
@@ -0,0 +1,379 @@
+Some nodes in my cluster often go offline for long periods of time, during which my job is unable to make progress -
+replacement pods will not be scheduled as long as the node remains offline. I'd like for my jobs to be automatically unblocked
+as soon as possible, so I can make best use of the available resources (since the training process is synchronized,
+there is no risk of data corruption in my case).
+
+### Risks and Mitigations
+
+#### Replacing Pods Which Are Still Running
+
+Starting a replacement pod without a guarantee that the original one was properly terminated carries an inherent risk
+and might result in inconsistencies, as discussed above.
+If this risk is not communicated to the user properly and the feature is not sufficiently explicit in its behavior,
+the users might mistake it for a general issue within Kueue/the broader Kubernetes ecosystem.
+
+As this risk stems from the behavior of the `kubelet` and control plane themselves, it cannot be fully mitigated without major changes in how Kubernetes operates.
+Instead, it should be adequately documented to prevent users without a compatible use-case from using it.
+Moreover, enabling this feature should require opt-in both from:
+1. The Administrator - by enabling the feature globally.
+2. The User - by opting specific workloads into the failure recovery mechanism.
+
+Controlling the covered workloads, instead of applying the recovery globally, will guarantee that only pods that were deemed "safe to forcefully terminate" are affected.
+
+## Design Details
+
+### Affected Pods
+
+In order to allow the first adopters to control which pods are affected by the new controller, a new `Pod` annotation will be introduced:
+```yaml
+kueue.x-k8s.io/safe-to-forcefully-terminate: "true"
+```
+
+Only pods containing the new annotation will be affected by the controller, for example:
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  generateName: sample-job-
+  namespace: default
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+spec:
+  parallelism: 3
+  completions: 3
+  podReplacementPolicy: Failed
+  template:
+    metadata:
+      annotations:
+        kueue.x-k8s.io/safe-to-forcefully-terminate: "true"  // <- new annotation
+    spec:
+      containers:
+      - name: dummy-job
+        image: registry.k8s.io/e2e-test-images/agnhost:2.53
+        command: [ "/bin/sh" ]
+        args: [ "-c", "sleep 600" ]
+        resources:
+          requests:
+            cpu: "1"
+            memory: "200Mi"
+      restartPolicy: Never
+```
+
+> The controller is not limited to pods managed by Kueue (i.e. having the `kueue.x-k8s.io/managed` or `kueue.x-k8s.io/podset` label). All pods in the cluster that have the new annotation will be affected.
+
+### Grace Period
+
+A default grace period of **1 minute** will be introduced to get initial feedback about the feature.
+This will help inform the decision on the structure of the API and setting defaults makes sense when graduating to beta.
```

Comment:
Can you mention that the grace period is counted based on Pod `.metadata.deletionGracePeriodSeconds`?